src/RFC-0122_Burning.md

```diff
@@ -0,0 +1,252 @@
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+## Language
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", 
+"NOT RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in 
+[BCP 14](https://tools.ietf.org/html/bcp14) (covering RFC2119 and RFC8174) when, and only when, they appear in all capitals, as 
+shown here.
+
+## Disclaimer
+
+This document and its content are intended for information purposes only and may be subject to change or update
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community of the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+The aim of this Request for Comment (RFC) is to describe the process of Burning UTXOs and to track the amount burned. 
+
+## Related Requests for Comment
+
+* [RFC-0100: Base Layer](RFC-0100_BaseLayer.md)
+* [RFC-0120: Consensus](RFC-0120_Consensus.md)
+
+## Description
+
+Blockchains have used the burn method to destroy coins in circulation and block their use forever. Most chains use a reclaimable address
+to denote burned coins. Using [RFC-0201: TariScript](RFC-0201_TariScript.md), Tari can also use this method, but this does not explicitly
+remove the coins from circulation. This RFC details a method to remove coins from circulation permanently.
+
+
+## Introduction
+
+To completely remove coins from circulation, we need to mark outputs as burned, and we need to change the balance equation to
+allow pruned nodes and non pruned nodes to still verify the integrity and emission of Tari.
+
+### Transaction output changes
+
+Currently, each transaction output has a field called [OutputFeatures] that tracks the unique properties of each output. 
+Inside this field, we track every possible type of output. We need to add another type here called burned. 
+
+```rust,ignore
+pub enum OutputType {
+    /// An standard non-coinbase output.
+    Standard = 0,
+    /// Output is a coinbase output, must not be spent until maturity.
+    Coinbase = 1,
+    /// This output is burned and may never be used as a transaction input.
+    Burned = 2,
+}
+```
+
+### Blockheader changes
+
+On the header of each block, we need to add a field for tracking the sum of all burned outputs on that block.
+
+```rust,ignore
+pub struct BlockHeader {
+    /// Version of the block
+    pub version: u16,
+    ...
+    /// Sum of kernel offsets for all kernels in this block.
+    pub total_kernel_offset: BlindingFactor,
+    /// Sum of script offsets for all kernels in this block.
+    pub total_script_offset: BlindingFactor,
+    /// Sum of all burned output commitments in this block.
+    pub burned_total: Commitment,
```

Comment:
I would actually propose making this the sum of burnt outputs in this block PLUS the value from the previous header (or ZERO if it's the genesis block). i.e. a running total of all BTXOs

Reply:
I don't really have a preference here, so happy to go with that as well

Reply:
Any reason for this?
I thought about doing it that way but decided against it due to the fact of how we approach and validate blocks in code currently. 
When validating a block, we don't look at overall balance, we only do validation on the block. 
for example:
Is the reward correct for this block,
does this block balance, etc. 
We never do chain balance validation etc, we make the assumption that if the currently stored state is correct we only have to validate the delta and not the entire state again. We do the same with the offsets. This follows that mindset to only validate the delta.

Reply:
We do `parent = prev.hash` and `height = prev.height + 1`, as well as checking MMR roots so I don't think there's much extra work in validating the running total vs total for the block.

Reply:
Don't get me wrong, it's definitely doable, 
This way just fits in better with the current validation and header checks. 
Hence why I asked if there is specifically a reason why we want to do it this way?